src/CommunityToolkit.Aspire.Hosting.Ollama/OllamaModelResource.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
+
+namespace Aspire.Hosting.ApplicationModel;
+
+/// <summary>
+/// A resource that represents an ollama model.
+/// </summary>
+/// <param name="name">The name for the resource.</param>
+/// <param name="modelName">The name of the LLM model, can include a tag.</param>
+/// <param name="parent">The <see cref="OllamaResource"/> parent.</param>
+public class OllamaModelResource(string name, string modelName, OllamaResource parent) : Resource(name), IResourceWithParent<OllamaResource>, IResourceWithConnectionString
+{
+    public OllamaResource Parent { get; } = ThrowIfNull(parent);
+
+    /// <summary>
+    /// Gets the connection string expression for the Ollama model.
+    /// </summary>
+    public ReferenceExpression ConnectionStringExpression => ReferenceExpression.Create($"{Parent}?Model={ModelName}");
```

Comment:
This is getting me to rethink the "connection string" concept that we have and whether we should go to a "real" connection string like `Endpoint=...;Model=...` structure.

That would be a breaking change but it does align better with how other Aspire integrations work (and we can use the `DbConnectionStringBuilder` class to parse it).